samples/retained-instance-store/src/main/java/com/badoo/ribs/samples/retained_instance_store/rib/retainedInstance/RetainedInstancePresenter.kt

```diff
@@ -0,0 +1,43 @@
+package com.badoo.ribs.samples.retained_instance_store.rib.retainedInstance
+
+import android.os.Parcelable
+import androidx.lifecycle.Lifecycle
+import com.badoo.ribs.android.subscribe
+import com.badoo.ribs.core.plugin.ViewAware
+import com.badoo.ribs.routing.source.backstack.BackStack
+import com.badoo.ribs.routing.source.backstack.operation.replace
+import com.badoo.ribs.samples.retained_instance_store.rib.retainedInstance.routing.RetainedInstanceRouter
+import java.lang.IllegalStateException
+
+interface RetainedInstancePresenter {
+    fun toggleChildCounter()
+}
+
+internal class RetainedInstancePresenterImpl(
+        private val backStack: BackStack<RetainedInstanceRouter.Configuration>
+) : ViewAware<RetainedInstanceView>,
+        RetainedInstancePresenter {
+
+    private var view: RetainedInstanceView? = null
+
+    override fun onViewCreated(view: RetainedInstanceView, viewLifecycle: Lifecycle) {
+        super.onViewCreated(view, viewLifecycle)
+        viewLifecycle.subscribe(
+                onCreate = { this.view = view },
+                onDestroy = { this.view = null }
+        )
+    }
+
+    override fun toggleChildCounter() {
+        val currentConfig = backStack.getCurrentConfig()
+        val newConfig = if (currentConfig is RetainedInstanceRouter.Configuration.Retained) {
+            RetainedInstanceRouter.Configuration.NotRetained
+        } else {
+            RetainedInstanceRouter.Configuration.Retained
+        }
+        backStack.replace(newConfig)
+    }
+
+    private fun <T : Parcelable> BackStack<T>.getCurrentConfig() = requireNotNull(state.current?.routing?.configuration) { IllegalStateException() }
```

Comment:
Not needed! You have backStack.activeConfiguration property provided by the framework.

Reply:
I don't get how that property can help me here, observing the state and storing it in the presenter? Wouldn't this be cleaner?

Reply:
You're right, this is a stream, my bad. We'll need a current value too. Can you just please add this comment as a reminder.

```suggestion
    private fun <T : Parcelable> BackStack<T>.getCurrentConfig() = 
        // FIXME add synchronous property to BackStack too
        requireNotNull(state.current?.routing?.configuration) { IllegalStateException() } 
```